classes/external/get_data_merge_requests.php

```diff
@@ -0,0 +1,150 @@
+        return new external_function_parameters([
+            'mergeusers' => new external_single_structure([
+                'removeuserfield' => new external_value(
+                    PARAM_TEXT, 'Remove user field', VALUE_OPTIONAL),
+                'removeuservalue' => new external_value(
+                    PARAM_RAW, 'Remove user value', VALUE_OPTIONAL),
+                'removeuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the remove user id', VALUE_OPTIONAL),
+                'keepuserfield' => new external_value(
+                    PARAM_TEXT, 'Keep user field', VALUE_OPTIONAL),
+                'keepuservalue' => new external_value(
+                    PARAM_RAW, 'Keep user value', VALUE_OPTIONAL),
+                'keepuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the keep user id', VALUE_OPTIONAL),
+                'id' => new external_value(
+                    PARAM_INT, 'Identifier of the merge request', VALUE_OPTIONAL),
+                'status' => new external_value(
+                    PARAM_INT, 'Status of the merge request', VALUE_OPTIONAL)
+            ])
+        ]);
+    }
+    /**
+     * Return data of the custom queue of the merge request
+     *
+     * @param $mergeusers id of the merge users request
+     *
+     * Return array with data of the custom queue of the merge request
+     * @param $id
+     * @param $removeuserfield Remove user field
+     * @param $removeuserid Remove user id
+     * @param $removeuservalue Remove user value
+     * @param $keepuserfield Keep user field
+     * @param $keepuserid Keep user id
+     * @param $keepuservalue Keep user value
+     * @param $timeadded Time creation
+     * @param $timemodified Time modified
+     * @param $status Status
+     * @param $retries Number of retries
+     * @param $log Log
+     * 
+     */
+    public static function execute(array $mergeusers) {
+        global $DB;
+        // Validate all of the parameters.
+        $params = array();
+        $params = self::validate_parameters(self::execute_parameters(),
+                                                array('id' => $mergeusers->id));
+        $tablemr = merge_request::TABLE_MERGE_REQUEST;
+        $sql = "SELECT
+                    id, removeuserfield, removeuservalue, removeuserid,
+                    keepuserfield, keepuservalue, keepuserid,
+                    timeadded, timemodified, status, retries, log
+                FROM
+                   {".merge_request::TABLE_MERGE_REQUEST."}
+                WHERE
+                     (1=1) ";
+        if (isset($mergeusers['removeuserfield']) && !empty($mergeusers['removeuserfield'])) {
+            $sql = $sql." AND removeuserfield = ?";
+            array_push($params, $mergeusers['removeuserfield']);
+        }
+        if (isset($mergeusers['removeuservalue']) && !empty($mergeusers['removeuservalue'])) {
+            $sql = $sql." AND removeuservalue = ?";
```

Comment:
Sorry, I forgot to say here that every comparison of values should use the API `$DB->get_in_or_equals()`. This will ensure any comparison works properly in any Moodle supported database engine. Review this file and any other similar comparison on other SQL sentences, please.

Reply:
can you make an example using this API function ? thanks

Reply:
You can look for its use in Moodle core. It has to be easy.

Reply:
I searched via command line a moodle core this function with -->  grep -ir "get_in_or_equals()" but nothing was returned